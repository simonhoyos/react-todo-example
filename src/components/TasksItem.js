import React from 'react';
import { TasksConsumer } from '../Context';

export function TaskItem({ id, title, done }) {
  return (
    <TasksConsumer>
      {({ handleEdit, handleToggleTask }) => (
        <li>
          {title} - {done ? 'completa' : 'incompleta'}
          <button onClick={() => handleEdit(id, title, done)}>Editar</button>
          <button onClick={() => handleToggleTask(id)}>Completar</button>
        </li>
      )}
    </TasksConsumer>
  );
}
