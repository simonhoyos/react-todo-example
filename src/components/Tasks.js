import React from 'react';
import { TaskItem } from './TasksItem';

export function Tasks({ tasks, handleEdit, handleToggleTask }) {
  return (
    <ul>
      {tasks.length > 0 && tasks.map(({ id, title, done }) => {
        return (
          <TaskItem
            key={id}
            id={id}
            title={title}
            done={done}
            handleEdit={handleEdit}
            handleToggleTask={handleToggleTask}
          />
        );
      })}
    </ul>
  );
}
