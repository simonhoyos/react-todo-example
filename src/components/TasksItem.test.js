import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Tasks } from './Tasks';
import { TaskItem } from './TasksItem';



describe('<Tasks />', () => {
  it('should match snapshot', () => {
    const taskMock = { id: 0, title: 'una tarea', done: true };
    const wrapper = shallow(<TaskItem {...taskMock} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should show incomplete when task is undone', () => {
    const taskMock = { id: 0, title: 'una tarea', done: false };
    const wrapper = shallow(<TaskItem {...taskMock} />);

    const item = wrapper.find('li');
    expect(item.props().children).toContain('incompleta');
  });

  it('should execute handleEdit when task is click', () => {
    const handleEdit = jest.fn();
    const taskMock = { id: 0, title: 'una tarea', done: false };
    const wrapper = shallow(<TaskItem {...taskMock} handleEdit={handleEdit} />);

    const editButton = wrapper.find('[data-testid="edit-button"]');
    editButton.simulate('click');

    expect(handleEdit).toHaveBeenCalled();
  });

  it('should execute handleToggleTask when task is click', () => {
    const handleToggleTask = jest.fn();
    const taskMock = { id: 0, title: 'una tarea', done: false };
    const wrapper = shallow(<TaskItem {...taskMock} handleToggleTask={handleToggleTask} />);

    const editButton = wrapper.find('[data-testid="toggle-button"]');
    editButton.simulate('click');

    expect(handleToggleTask).toHaveBeenCalled();
  });
});
