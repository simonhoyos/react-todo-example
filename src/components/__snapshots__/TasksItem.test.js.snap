// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tasks /> should match snapshot 1`] = `
<li>
  una tarea
   - 
  completa
  <button
    data-testid="edit-button"
    onClick={[Function]}
  >
    Editar
  </button>
  <button
    data-testid="toggle-button"
    onClick={[Function]}
  >
    Completar
  </button>
</li>
`;
